### ISSUE TITLE
Convergence test for gemma2

### ISSUE THREAD

**Issue by Tcc0403** [linkedin/Liger-Kernel#370]
### 🐛 Describe the bug

Currently convergence test doesn't include gemma2 because it cannot pass the test. After investigating, it seems that LigerRMSNorm is the cause. 

### Reproduce

Simply set [`rms_norm=False`](https://github.com/linkedin/Liger-Kernel/blob/2d3eb94679cec946c8f85a15590cbef9c1327250/test/convergence/test_mini_models.py#L398) (not patching it) can make gemma2 pass the convergence test, `test_mini_models.py`. 

Here's the output if setting `rms_norm=True` and my assumptions.
I set [`num_step=2`](https://github.com/linkedin/Liger-Kernel/blob/2d3eb94679cec946c8f85a15590cbef9c1327250/test/convergence/test_mini_models.py#L639) and [print logits and its grad out](https://github.com/linkedin/Liger-Kernel/blob/2d3eb94679cec946c8f85a15590cbef9c1327250/test/convergence/test_mini_models.py#L438) for observation only.
```
Liger kernel patches have been reverted.
Step 0, Loss: 6.3974809646606445
Step 0, logits: tensor([ 0.0000,  0.6915,  4.3167,  ..., -0.1310,  0.4010,  0.3292],
       device='cuda:0', grad_fn=<SelectBackward0>)
Step 0, logits's grads: tensor([ 1.2537e-08,  2.5032e-08, -4.9119e-04,  ...,  1.0997e-08,
         1.8721e-08,  1.7424e-08], device='cuda:0')
Step 1, Loss: 0.950512707233429
Step 1, logits: tensor([-0.0451,  0.3177, 11.4745,  ...,  0.4984, -0.2665,  1.2135],
       device='cuda:0', grad_fn=<SelectBackward0>)
Step 1, logits's grads: tensor([ 3.5163e-09,  5.0539e-09, -1.3815e-04,  ...,  6.0553e-09,
         2.8179e-09,  1.2379e-08], device='cuda:0')
Liger kernel patches have been reverted.
Liger kernel patches have been reverted.
Step 0, Loss: 6.39747953414917
Step 0, logits: tensor([ 0.0000,  0.6915,  4.3167,  ..., -0.1310,  0.4010,  0.3292],
       device='cuda:0', grad_fn=<SelectBackward0>)
Step 0, logits's grads: tensor([ 1.2537e-08,  2.5032e-08, -4.9119e-04,  ...,  1.0997e-08,
         1.8721e-08,  1.7424e-08], device='cuda:0')
Step 1, Loss: 0.7350322008132935
Step 1, logits: tensor([-0.0521,  0.0941, 12.8524,  ...,  0.2240, -0.0294,  0.7218],
       device='cuda:0', grad_fn=<SelectBackward0>)
Step 1, logits's grads: tensor([ 1.1150e-09,  1.2904e-09, -4.3773e-05,  ...,  1.4696e-09,
         1.1405e-09,  2.4174e-09], device='cuda:0')
Liger kernel patches have been reverted.
```
In step 0, there are no difference between patching and non-patching, so I think forward pass is fine.
In step 1, logits are off, which means parameters weren't correctly updated, ie backward pass might go wrong.

Everything is done in fp32, so I **_guess_** casting method is not the main reason.

### Versions

Environment Report:
-------------------
Operating System: Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python version: 3.10.12
PyTorch version: 2.4.1+cu121
CUDA version: 12.1
Triton version: 3.0.0
Transformers version: 4.45.2

**Comment by Tcc0403** [comment]
update: inplace operations on dY will cause issues for [`Gemma2DecodeLayer.post_attention_layernorm` and `Gemma2DecodeLayer.post_feedforward_layernorm`](https://github.com/huggingface/transformers/blob/3ea3ab62d80d91f9bdd16bd3cacd8133fb0d4566/src/transformers/models/gemma2/modeling_gemma2.py#L496-L507)
https://github.com/linkedin/Liger-Kernel/blob/b2b6970afea80e43ad20c005926f65d8cc0d309e/src/liger_kernel/ops/rms_norm.py#L268-L270
create new tensor for dX can avoid this problem
